Client/Frontend/Browser/Favicons/FaviconFetcher.swift

```diff
@@ -89,22 +93,40 @@ class FaviconFetcher {
     ///        downloaded and parsed for a favicon.
     ///     4. Monogram (letter + background color)
     func load(_ cachedOnly: Bool = false, _ completion: @escaping (URL, FaviconAttributes) -> Void) {
-        if let icon = customIcon {
-            completion(url, icon)
-            return
-        }
-        let matchesFetchKind = faviconOnFileMatchesFetchKind(domain.favicon)
-        if let icon = bundledIcon, domain.favicon == nil || !matchesFetchKind {
-            completion(url, icon)
-            return
-        }
-        fetchIcon(cachedOnly) { url, attributes in
-            DispatchQueue.main.async {
-                completion(url, attributes)
+        let faviconType = domain.favicon?.type
+        
+        loadTaskCancellable = DispatchWorkItem(qos: .utility, block: { [weak self] in
+            guard let self = self else { return }
+            if let icon = self.customIcon {
+                DispatchQueue.main.async {
+                    completion(self.url, icon)
+                }
+                return
+            }
+            let matchesFetchKind = self.faviconOnFileMatchesFetchKind(faviconType)
+            if let icon = self.bundledIcon, faviconType == nil || !matchesFetchKind {
+                DispatchQueue.main.async {
+                    completion(self.url, icon)
+                }
+                return
+            }
+            self.fetchIcon(cachedOnly) { url, attributes in
+                DispatchQueue.main.async {
+                    completion(url, attributes)
+                }
             }
+        })
+        
+        if let task = loadTaskCancellable {
+            DispatchQueue.global(qos: .utility).async(execute: task)
```

Comment:
`utility` is a low priority task, something like seeing an icon may need to be bumped up to `userInitiated`. This also may spawn a lot of queue's on the global queue at once, are we sure we don't want to use a private `NSOperationQueue` instead?